### DISCUSSION TITLE
Is there a solution to use a `HeartbeatSender` with lazy AMQP stream connections?

### DISCUSSION THREAD

**Discussion by mrtus** [php-amqplib/php-amqplib#1091] [Q&A] [answered]
👋 

We want to enable heartbeat sending in processes which launch with a lazy AMQP connection, however the sender requires a connection to be already connected.

Is there a callback available somewhere that could trigger the heartbeat sender to be created after connection?

Most of those processes are very short lived and do not require the heartbeat sender connection, however some do result in longer processes, where we do want the heartbeat sender to be enabled.

Of course always connecting to AMQP when the connection would not even be used is unreasonable.

Thanks!

**Comment by mrtus** [comment, marked as answer]
I guess we can close this one since apparently I asked this question again later and also solved it myself 🤦 

See https://github.com/php-amqplib/php-amqplib/issues/1099

**Reply by lukebakken** [reply]
Thank you!
